src/components/ContactsList/_modals/ContactUpsertModal/ContactEditModal.tsx

```diff
@@ -0,0 +1,34 @@
+import { FC, useEffect } from 'react';
+import { ContactUpsertModal } from './ContactUpsertModal';
+import { useContactFields } from './useContactFields';
+import { Contact } from '@/api';
+
+type ContactEditModalProps = {
+  contact: Contact;
+  onSubmitted: () => void;
+  onCancel: () => void;
+};
+
+export const ContactEditModal: FC<ContactEditModalProps> = ({ onSubmitted, onCancel, contact }) => {
+  const fields = useContactFields();
+  const { setPaymail, setName, setPhone } = fields;
+
+  useEffect(() => {
+    setPaymail(contact.paymail);
+    setName(contact.fullName);
+    setPhone(contact?.metadata?.phoneNumber ?? '');
+  }, [contact, setName, setPaymail, setPhone]);
+
+  return (
+    <ContactUpsertModal
+      onCancel={onCancel}
+      onSubmitted={onSubmitted}
+      sucessMsg="Contact edited successfully!"
+      errorMsg="Error occurred while editing a contact."
+      modalTitle="Edit contact"
+      modalSubtitle=""
+      fields={fields}
+      disabledPaymailInput={true}
+    />
+  );
+};
```

Comment:
If pass `contact` into `ContactUpsertModal` then it possible to move `useEffect` logic into `ContactUpsertModal`. Then call `useContactFields` in `ContactUpsertModal`

Reply:
No, because I applied here some kind of "compose strategy". 
The `ContactUpsertModal` is used also in `ContactAddModal` component.